api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -162,38 +164,56 @@ trait InstitutionsHttpApi extends InstitutionProtocol with ApiErrorProtocol with
   val uploadPath =
     path("institutions" / Segment / "filings" / Segment / "submissions" / Segment) { (institutionId, period, submissionId) =>
       val path = s"institutions/$institutionId/filings/$period/submissions/$submissionId"
-      val uploadTimestamp = Instant.now.toEpochMilli
-      val processingActor = createHmdaRawFile(system, submissionId)
-      processingActor ! StartUpload
-      fileUpload("file") {
-        case (metadata, byteSource) if (metadata.fileName.endsWith(".txt")) =>
-          time {
-            val uploadedF = byteSource
-              .via(splitLines)
-              .map(_.utf8String)
-              .runForeach(line => processingActor ! AddLine(uploadTimestamp, line))
-
-            onComplete(uploadedF) {
-              case Success(response) =>
-                processingActor ! CompleteUpload
-                processingActor ! Shutdown
-                complete {
-                  "uploaded"
-                }
-              case Failure(error) =>
-                processingActor ! Shutdown
-                log.error(error.getLocalizedMessage)
-                val errorResponse = ErrorResponse(400, "Invalid File Format", path)
-                complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
-            }
-          }
+      extractExecutionContext { executor =>
+        val uploadTimestamp = Instant.now.toEpochMilli
+        val processingActor = createHmdaRawFile(system, submissionId)
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(institutionId, period))
+        implicit val ec: ExecutionContext = executor
+        val isNotOverwrite = preventSubmissionOverwrite(submissionsActor, submissionId.toInt)
+        onComplete(isNotOverwrite) {
+          case Success(isNotOverwrite) =>
```

Comment:
I found a way to remove one layer of nesting, so the logic doesn't get so deep.

Instead of having an `if`/`else` statement on the outcome of `isNotOverwrite`, you can use `case` statements like this:

```scala
onComplete(preventSubmissionOverwrite(submissionsActor, submissionId.toInt)) {
          case Success(false) =>
            val errorResponse = ErrorResponse(400, "Submission already exists", path)
            complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
          case Success(true) =>
            //upload the file
          case Failure(_) =>
            submissionsActor ! Shutdown
            val errorResponse = ErrorResponse(500, "Internal server error", path)
            complete(ToResponseMarshallable(StatusCodes.InternalServerError -> errorResponse))
        }
```

... if you use this strategy, you might want to update the name of the `preventSubmissionOverwrite` function to indicate what `true